File: dotnet/src/PuzzleSolver.NumbersGame/BoardRules.cs
namespace PuzzleSolver.NumbersGame;

public static class BoardRules
{
    internal const int StartingSize = 6;

    public static int ReuseLimit(Number number)
    {
        return BoardRules.ReuseLimit(number.Category);
    }

    public static int ReuseLimit(NumberCategory category)
    {
        return category switch
        {
            NumberCategory.Large => 1,
            NumberCategory.Small => 2,
            _ => int.MaxValue
        };
    }
}

File: python/board_rules.py
import sys

from number import Number, NumberCategory


class _BoardRules:
    @staticmethod
    def starting_size() -> int:
        """
        Returns the starting size of the board.

        :return: The starting size of the board.
        """
        return 6

    @staticmethod
    def reuse_limit(number: Number) -> int:
        """
        Determines the reuse limit for a number based on its category.

        :param number: The Number instance to check.
        :return: The maximum number of times this number can be reused.
        """
        if number.category == NumberCategory.SMALL:
            return 2
        elif number.category == NumberCategory.LARGE:
            return 1
        else:
            return sys.maxsize
